src/pages/Environments.js

```diff
@@ -39,13 +39,14 @@ const DeleteRuntimeModal = ({
 }) => {
   const [deleteDisk, setDeleteDisk] = useState(false)
   const [deleting, setDeleting] = useState()
+  const ajax = useReplaceableAjax()
   const deleteRuntime = _.flow(
     Utils.withBusyState(setDeleting),
     withErrorReporting('Error deleting cloud environment')
   )(async () => {
     isGcpContext(cloudContext) ?
-      await Ajax().Runtimes.runtime(googleProject, runtimeName).delete(deleteDisk) :
-      await Ajax().Runtimes.runtimeV2(workspaceId, runtimeName).delete(deleteDisk)
+      await ajax().Runtimes.runtime(googleProject, runtimeName).delete(deleteDisk) :
```

Comment:
I recommend a more DI-like solution for ajax that doesn't rely on react context.  Similar to my suggestion for nav.getlink dependency, but more detailed... I propose something along these lines (specified in TS for clarity):
```
export type AjaxContract = ReturnType<typeof Ajax>;
export type AjaxRuntimesContract = AjaxContract['Runtimes'] // could be narrowed further with Pick<> on only used methods

export type AjaxRuntimesFactory = (signal?: AbortSignal) => AjaxRuntimesContract

...
export interface EnvironmentPageAjax {
  runtimes: AjaxRuntimesFactory
  ... // list factories for just the 3 sub-areas the component is using
}
export interface EnvironmentPageProps {
  ...
  ajax:  EnvironmentPageAjax
}
```
in the component:
`ajax.runtimes(signal).theDataCall(...)`

in EnvironmentsPage wrapper component:
```
h(Enviroments, {
  ajax: {
    runtimes: (signal) => Ajax(signal).Runtimes,
    ...
  }
}  
```
A similar DI pattern can be employed to make useWorkspaces a factory-driven DI with composition here in Terra UI that enables recomposition with AoU-friendly dependency swap-ins.  This would make the shared contract more explicit even now in this iterative solution.

I think this would work a lot better if the component file and workspace-utils.js get a minimum conversion to Typescript as well... probably a discussion for the mob session.

Reply:
oops, I mean to add this comment to the Environments component's ajax usage, but the pattern would need replicating here too.

Reply:
I believe I would have an easier time parsing plain prose. What I think you are suggesting is that I narrow the scope of the Ajax object to include only the attributes that are actually used within the component. If that is your suggestion, then I don't see how to do that without significant modifications to the file's structure, since many of the components depend directly on the Ajax object as imported.

Reply:
lets defer this larger challenge of how we deal with the broadness of Ajax to a later time.